packages/backend/src/models/CatalogModel/utils/parser.ts

```diff
@@ -31,8 +31,8 @@ export const parseFieldsFromCompiledTable = (
     table: CompiledTable,
 ): CatalogField[] => {
     const tableFields = [
-        ...Object.values(table.dimensions),
-        ...Object.values(table.metrics),
+        ...Object.values(table.dimensions).filter((d) => !d.hidden),
```

Comment:
mm, I thought we were doing this already. Was this perhaps disabled to allow `filtering` in the UI ? @magnew